charts/linkerd2/templates/prometheus.yaml

```diff
@@ -219,6 +232,12 @@ spec:
         securityContext:
           runAsUser: 65534
         volumeMounts:
+        {{- range .Values.prometheusExtraConfigmapMounts }}
+          - name: {{ .name }}
+            mountPath: {{ .mountPath }}
+            subPath: {{ .subPath }}
+            readOnly: {{ .readOnly }}
+        {{- end }}
```

Comment:
The extra indentation is breaking the configmap.
Also, since `rule_files` is looking under `/etc/prometheus`, we always want to mount in that dir, so we can get rid of the `mountPath` parameter under `.Values.prometheusExtraConfigmapMounts` and use `.subPath` here:
```yaml
mountPath: /etc/prometheus/{{ .subPath }}
subPath: {{ .subPath }}
```
I also think we can hard-code `readOnly: true`.

I believe this also requires changing the main config mountPath below to something like:
```yaml
- mountPath: /etc/prometheus/prometheus.yml
  subPath: prometheus.yml
  name: prometheus-config
  readOnly: true
```

Reply:
This sgtm provided that we ain't gonna need configmaps mounted other than rules files. Do you see in the near future any cause for a user wanting to mount a config map to prometheus other than rules files/elsewhere than /etc/prometheus?

Reply:
That's a good point. People might wanna mount a CM for setting environment variables for authenticating against say GCE for service discovery. So yeah, I agree with the renaming you propose :+1:

Reply:
Done!